docs/content/_navigation.json

```diff
@@ -123,7 +123,21 @@
         "children": [
           {
             "title": "Partitions",
-            "path": "/concepts/partitions-schedules-sensors/partitions"
+            "path": "/concepts/partitions-schedules-sensors/partitions",
```

Comment:
one practical question is as we got rid of the links on sub-section titles, is it a good idea to put a link here? i.e.
this doesn't have a link:
<img width="300" alt="image" src="https://github.com/dagster-io/dagster/assets/4531914/6a6af627-a88e-430b-bf09-c56e02b70015">
while this links to the /partitions page:
<img width="300" alt="image" src="https://github.com/dagster-io/dagster/assets/4531914/12384c14-6168-4fd5-b7a7-5637980c1402">

Reply:
Good catch - yes, I'll add it. Links on subpages were removed before because they were redundant (ex: `Concepts > Resources & Config` pointed to `/concepts/resources`, but so did `Concepts > Resources > Resources`. As this concept now has a dedicated category page, I think it'll be good to link it here.

Reply:
I think it's actually okay - subpage links were removed before because they were redundant or a suitable category page didn't exist. Ex: `Concepts > Resources & Config` linked to `/concepts/resources`, but so did `Concepts > Resources & Config > Resources`.

I think adding leaving `Partitions & Backfills` as it is is fine for now. We could add a category page for this that talks about partitions and backfills together - similar to how the `/concepts/partitions-schedules-backfills/partitions` does for partitions in this PR - but it might be a bit redundant if we also have a dedicated backfills page.

Reply:
ok leaving it as in sounds good to me. we can follow up the hierarchy improvement later.